libs/components/forms/src/lib/modules/character-counter/character-counter.directive.ts

```diff
@@ -31,36 +31,38 @@ export class SkyCharacterCounterInputDirective implements Validator {
    * `textarea` element.
    */
   @Input()
-  public skyCharacterCounterIndicator: SkyCharacterCounterIndicatorComponent;
+  public skyCharacterCounterIndicator:
+    | SkyCharacterCounterIndicatorComponent
+    | undefined;
 
   /**
    * Specifies the maximum number of characters allowed in the input field. Place this directive
    * on an `input` or `textarea` element. This property accepts `number` values.
    * @required
    */
   @Input()
-  public set skyCharacterCounterLimit(value: number) {
-    this._skyCharacterCounterLimit = value;
-    this.updateIndicatorLimit(this.skyCharacterCounterLimit);
-    this._validatorChange();
-  }
-
-  public get skyCharacterCounterLimit(): number {
-    return this._skyCharacterCounterLimit || 0;
+  public set skyCharacterCounterLimit(value: number | undefined) {
+    if (value === undefined) {
+      this.#skyCharacterCounterLimitOrDefault = 0;
+    } else {
+      this.#skyCharacterCounterLimitOrDefault = value;
+    }
+    this.#updateIndicatorLimit(this.#skyCharacterCounterLimitOrDefault);
+    this.#_validatorChange();
   }
 
-  private _skyCharacterCounterLimit: number;
+  #skyCharacterCounterLimitOrDefault = 0;
 
   public validate(control: AbstractControl): ValidationErrors {
     const value = control.value;
 
-    this.updateIndicatorCount((value && value.length) || 0);
+    this.#updateIndicatorCount((value && value.length) || 0);
 
     if (!value) {
-      return;
+      return {};
```

Comment:
The `Validator` class's `validate` method returns a type of `ValidationErrors | null`, so you can return `null` here.

Reply:
Done